### PR TITLE
[Issue 9789] Add unit tests for formatCurrencyUtil

### LINKED ISSUE #9789: Add a Jest unit test at `frontend/src/utils/formatCurrencyUtil.test.ts`
### Summary

Add a Jest unit test at `frontend/src/utils/formatCurrencyUtil.test.ts` for the pure function `formatCurrency` in `frontend/src/utils/formatCurrencyUtil.ts`. The function wraps `Intl.NumberFormat` with USD + 0 minimum fraction digits and returns an empty string on a null or falsy input. It has no sibling test today.

## Background & Context
Of the utilities under `frontend/src/utils/`, several files have tests (`dateUtil.test.ts`, `generalUtils.test.ts`) but `formatCurrencyUtil.ts` does not. The existing test files follow a consistent structure that new contributors can copy.

Governing rule: `.cursor/rules/frontend-tests.md` (Jest + RTL conventions) and the utility-test pattern in `frontend/src/utils/dateUtil.test.ts`.

## Getting Started
```
cd frontend
npm install
npm run test -- formatCurrencyUtil
```
Reference `frontend/src/utils/dateUtil.test.ts` for the describe/it structure.

## Technical Approach
Source (`frontend/src/utils/formatCurrencyUtil.ts:1-10`):
```ts
export const formatCurrency = (numberToFormat: number | null) => {
  if (numberToFormat) {
    return new Intl.NumberFormat("en-US", {
      style: "currency",
      currency: "USD",
      minimumFractionDigits: 0,
    }).format(numberToFormat);
  }
  return "";
};
```
Test:
```ts
import { formatCurrency } from "src/utils/formatCurrencyUtil";

describe("formatCurrency", () => {
  it("formats positive integers as USD", () => {
    expect(formatCurrency(1000)).toBe("$1,000");
  });
  it("formats large numbers with commas", () => {
    expect(formatCurrency(1234567)).toBe("$1,234,567");
  });
  it("returns empty string for zero (falsy input)", () => {
    expect(formatCurrency(0)).toBe("");
  });
  it("returns empty string for null", () => {
    expect(formatCurrency(null)).toBe("");
  });
  it("formats negative values", () => {
    expect(formatCurrency(-500)).toBe("-$500");
  });
});
```
Note: document the zero-returns-empty behaviour -- this is a direct consequence of the truthy check on line 2 and matches the current implementation.

### Acceptance criteria

- [ ] `frontend/src/utils/formatCurrencyUtil.test.ts` exists.
- [ ] Tests cover: positive number (`1000` -> `"$1,000"`), zero (returns `""` because of the truthy check), null (returns `""`), negative number (returns the `Intl` formatted string).
- [ ] `cd frontend && npm run test -- formatCurrencyUtil` passes.
- [ ] No modifications to `formatCurrencyUtil.ts` itself.

## Testing Requirements
- `cd frontend && npm run test -- formatCurrencyUtil` all pass.
- `cd frontend && npm run lint` produces no new warnings.

### PR DESCRIPTION
**Pull request by eddie333016**
## Summary
Adds comprehensive unit tests for the three exported functions in `formatCurrencyUtil.ts`:
- `formatCurrency` — USD formatting with null/falsy/positive/decimal handling
- `formatCurrencyString` — currency-like string parsing with graceful fallback for non-numeric input
- `getNumericAmountFromString` — extracting numeric values from comma-separated strings

Resolves #9789

## Testing
- `npm run test -- formatCurrencyUtil` passes
- Follows existing test patterns from `dateUtil.test.ts`

#### Reviewers
Tagging @btabaska @mdragon @KevinJBoyer per contributing guide.

### CODE REVIEW

File: frontend/src/utils/formatCurrencyUtil.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { formatCurrency, formatCurrencyString, getNumericAmountFromString } from "src/utils/formatCurrencyUtil";
+
```
Comment (btabaska) [inline frontend/src/utils/formatCurrencyUtil.test.ts]: Prettier isn't happy with this file (prettier --check fails), so the lint step in CI will block too. The import line is over our 80-char width and 
  ▎ needs to wrap. A quick npx prettier --write src/utils/formatCurrencyUtil.test.ts fixes it

**Comment by btabaska** [comment]
Once the suite is green and
  lint passes, this is good to go. The structure and edge-case coverage are solid. Happy to chat if anything here is unclear!

**Review by btabaska — CHANGES_REQUESTED** [review]

**Comment by btabaska** [comment]
I just went ahead and made the changes since they were minimal. Thank you so much for taking the time to contribute to simpler.grants.gov!

**Review by btabaska — APPROVED** [review]
